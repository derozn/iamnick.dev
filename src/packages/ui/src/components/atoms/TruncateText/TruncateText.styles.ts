'use client';

import { tv } from '@derozn/tailwind-variants';

import { twMergeConfig } from '@iamnick/ui/src/theme';

export const truncateTextStyles = tv(
  {
    slots: {
      base: '',
    },
    variants: {
      lines: {
        1: {
          base: 'ui-line-clamp-1',
        },
        2: {
          base: 'ui-line-clamp-2',
        },
        3: {
          base: 'ui-line-clamp-3',
        },
      },
    },
    defaultVariants: {
      lines: 2,
    },
  },
  {
    responsiveVariants: true,
    twMergeConfig,
  },
);
