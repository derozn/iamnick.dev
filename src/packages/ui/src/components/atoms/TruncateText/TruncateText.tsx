import { VariantProps } from '@derozn/tailwind-variants';
import { FC, ReactNode } from 'react';

import { truncateTextStyles } from './TruncateText.styles';

export interface ITruncateText extends VariantProps<typeof truncateTextStyles> {
  children?: ReactNode;
}

export const TruncateText: FC<ITruncateText> = ({ children, lines }) => {
  const { base } = truncateTextStyles({ lines });

  return <span className={base()}>{children}</span>;
};
