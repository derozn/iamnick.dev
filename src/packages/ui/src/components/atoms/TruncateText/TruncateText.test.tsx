import { render } from '@testing-library/react';

import { TruncateText, ITruncateText } from './TruncateText';

const setup = (props: ITruncateText) => {
  return render(<TruncateText {...props} />);
};

const lineCount = [1, 2, 3] as const;

describe(TruncateText.name, () => {
  describe('When rendering variants', () => {
    describe.each(lineCount)('and the line count %s', (lines) => {
      it('renders correctly', () => {
        const wrapper = setup({
          lines,
        });
        expect(wrapper.container.firstChild).toMatchSnapshot();
      });
    });
  });
});
