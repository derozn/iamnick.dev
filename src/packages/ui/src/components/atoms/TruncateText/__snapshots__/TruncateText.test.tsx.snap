// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TruncateText When rendering variants and the line count 1 renders correctly 1`] = `
<span
  class="ui-line-clamp-1"
/>
`;

exports[`TruncateText When rendering variants and the line count 2 renders correctly 1`] = `
<span
  class="ui-line-clamp-2"
/>
`;

exports[`TruncateText When rendering variants and the line count 3 renders correctly 1`] = `
<span
  class="ui-line-clamp-3"
/>
`;
